geomstats/geometry/quotient_manifold.py

```diff
@@ -0,0 +1,480 @@
+    base : Manifold
+        Base manifold of the bundle.
+        Optional. Default : None.
+    group : LieGroup
+        Group that acts on the total space by the right.
+        Optional. Default : None.
+        Either the group or the group action must be given.
+    group_action : callable
+        Right group action. It must take as input a point of the total space
+        and an element of the group, and return a point of the total space.
+    dim : int
+        Dimension of the base manifold.
+        Optional. If available the dimension of the base manifold is used,
+        or the difference between the dimension of the total space and the
+        group. Either dim, base or group must be given as input.
+    """
+
+    def __init__(self, total_space, base=None, group=None, group_action=None,
+                 dim=None, **kwargs):
+
+        if dim is None:
+            if base is not None:
+                dim = base.dim
+            elif group is not None:
+                dim = total_space.dim - group.dim
+            else:
+                raise ValueError('Either the base manifold, '
+                                 'its dimension, or the group acting on the '
+                                 'total space must be provided.')
+
+        super(FiberBundle, self).__init__(dim=dim, **kwargs)
+
+        self.base = base
+        self.total_space = total_space
+        self.group = group
+
+        if group_action is None and group is not None:
+            group_action = group.compose
+        self.group_action = group_action
+
+    def belongs(self, point, atol=1e-6):
+        """Evaluate if a point belongs to the base manifold.
+
+        Evaluate if a point belongs to the base manifold when it is given,
+        otherwise to the total space.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., dim]
+            Point to evaluate.
+        atol : float
+            Absolute tolerance.
+            Optional, default: 1e-6.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean evaluating if point belongs to the manifold.
+        """
+        if self.base is not None:
+            return self.base.belongs(point, atol=atol)
+        return self.total_space.belongs(point, atol=atol)
+
+    @staticmethod
+    def submersion(point):
+        """Project a point to base manifold.
+
+        This is the projection of the fiber bundle, defined on the total
+        space, with values in the base manifold. This map is surjective.
+        By default, the base manifold  is not explicit but is identified with a
+        local section of the fiber bundle, so the submersion is the identity
+        map.
+
+        Parameters
+        ----------
+        point: array-like, shape=[..., {ambient_dim, [n, n]}]
+            Point of the total space.
+
+        Returns
+        -------
+        projection: array-like, shape=[..., {dim, [n, n]}]
```

Comment:
NIT: maybe not use "projection" since no var is called projection here

Reply:
same as above